test/e2e/dns_configmap.go

```diff
@@ -0,0 +1,314 @@
+func (t *dnsConfigMapTest) runDig(dnsName string) []string {
+	cmd := []string{
+		"/usr/bin/dig",
+		"+short",
+		"@" + t.dnsPod.Status.PodIP,
+		"-p", "10053", dnsName,
+	}
+	stdout, stderr, err := t.f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     t.f.Namespace.Name,
+		PodName:       t.utilPod.Name,
+		ContainerName: "util",
+		CaptureStdout: true,
+		CaptureStderr: true,
+	})
+
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Running dig: %v, stdout: %q, stderr: %q",
+		cmd, stdout, stderr)
+
+	if stdout == "" {
+		return []string{}
+	} else {
+		return strings.Split(stdout, "\n")
+	}
+}
+
+func (t *dnsConfigMapTest) setConfigMap(cm *api.ConfigMap, isValid bool) {
+	if isValid {
+		t.cm = cm
+	}
+	t.isValid = isValid
+
+	cm.ObjectMeta.Namespace = t.ns
+	cm.ObjectMeta.Name = t.name
+
+	options := api.ListOptions{
+		FieldSelector: fields.Set{
+			"metadata.namespace": t.ns,
+			"metadata.name":      t.name,
+		}.AsSelector(),
+	}
+	cmList, err := t.c.Core().ConfigMaps(t.ns).List(options)
+	Expect(err).NotTo(HaveOccurred())
+
+	if len(cmList.Items) == 0 {
+		By(fmt.Sprintf("Creating the ConfigMap (%s:%s) %+v", t.ns, t.name, *cm))
+		_, err := t.c.Core().ConfigMaps(t.ns).Create(cm)
+		Expect(err).NotTo(HaveOccurred())
+	} else {
+		By(fmt.Sprintf("Updating the ConfigMap (%s:%s) to %+v", t.ns, t.name, *cm))
+		_, err := t.c.Core().ConfigMaps(t.ns).Update(cm)
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
+
+func (t *dnsConfigMapTest) deleteConfigMap() {
+	By(fmt.Sprintf("Deleting the ConfigMap (%s:%s)", t.ns, t.name))
+
+	t.cm = nil
+	t.isValid = false
+
+	err := t.c.Core().ConfigMaps(t.ns).Delete(t.name, nil)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func (t *dnsConfigMapTest) createUtilPod() {
+	t.utilPod = &api.Pod{
+		TypeMeta: unversioned.TypeMeta{
+			Kind: "Pod",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Namespace: t.f.Namespace.Name,
+			Name:      "e2e-dns-configmap-" + string(uuid.NewUUID()),
+			Labels:    map[string]string{"app": "e2e-dns-configmap"},
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:    "util",
+					Image:   "gcr.io/google_containers/dnsutils:e2e",
+					Command: []string{"sleep", "10000"},
+					Ports: []api.ContainerPort{
+						{ContainerPort: 10101, Protocol: "TCP"},
+					},
+				},
+			},
+		},
+	}
+
+	var err error
+	t.utilPod, err = t.c.Core().Pods(t.f.Namespace.Name).Create(t.utilPod)
+	Expect(err).NotTo(HaveOccurred())
+	framework.Logf("Created pod %v", t.utilPod)
+	Expect(t.f.WaitForPodRunning(t.utilPod.Name)).NotTo(HaveOccurred())
+
+	t.utilService = &api.Service{
+		TypeMeta: unversioned.TypeMeta{
+			Kind: "Service",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Namespace: t.f.Namespace.Name,
+			Name:      "e2e-dns-configmap",
+		},
+		Spec: api.ServiceSpec{
+			Selector: map[string]string{"app": "e2e-dns-configmap"},
+			Ports: []api.ServicePort{
+				{Protocol: "TCP", Port: 10101, TargetPort: intstr.FromInt(10101)},
```

Comment:
optional nit: you can make it a const at top and add a comment saying its random.